File: shortest-word-3.js
// Time, Space - O(N), O(1)
/**
 * @param {string[]} words
 * @param {string} word1
 * @param {string} word2
 * @return {number}
 */
var shortestWordDistance = function(words, word1, word2) {
  
    let j=-1, k=-1;
    let min = Number.MAX_VALUE;
    
    for(let i=0;i<words.length;i++) {
        let word = words[i];
        if(word === word1) {
            j = i;
        }
        if(word === word2) {
            if( j == i)  {
                j = k;
            }
            k = i;
        }
        if(j!= -1 && k != -1) {
            min = Math.min(min, Math.abs(j-k));
        }
    }
    
    return min;
    
};

File: shortestword3.py
class Solution:
    def shortestWordDistance(self, wordsDict: List[str], word1: str, word2: str) -> int:

        ind1, ind2, res = -1, -1, math.inf
        same = False
        if word1==word2:
            same = True
        for i in range(len(wordsDict)):
            if same and wordsDict[i]==word1:
                ind2 = ind1
                ind1 = i
            elif wordsDict[i]==word1:
                ind1 = i
            elif wordsDict[i]==word2:
                ind2 = i

            if ind2>=0 and ind1>=0:
                res = min(res, abs(ind1-ind2))

        return res

             
                